### PR TITLE
Integrate Redis for Core Data, Core Metadata, and Export Client

### PR DESCRIPTION
**Pull request by andresrinivasan**
Re issue #380. This PR includes 

1. the code written by @itamarhaber to integrate Redis into Core Data, Core Metadata, and Export Client.
1. updates for the new TOML based configuration
1. hooks in all the respective init.go
1. tested with as much of the data from the blackbox testing as possible
1. remaining tests were performed with data generated by the Go tests

### CODE REVIEW

File: internal/pkg/db/redis/export.go [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+
+import (
+	"github.com/edgexfoundry/edgex-go/internal/export"
+	"github.com/edgexfoundry/edgex-go/internal/pkg/db"
+	"github.com/gomodule/redigo/redis"
+	"gopkg.in/mgo.v2/bson"
+)
+
+const (
+	EXPORT_COLLECTION = "exportConfiguration"
```
[the remark this answers is not in the document]
Comment (andresrinivasan) [reply internal/pkg/db/redis/export.go]: Good catch. Corrected gaf in internal/pkg/db/redis/export.go.

The changes to README.md and glide.yaml are specific to the addition of Redis. I believe those make sense leaving in this branch. I will separate out .gitignore, bin/edgex-docker-launch.sh into separate issues/PRs.

Wrt the signoff, I need help. I cannot figure out how to sign off on the specific commit where its missing. I go through the amend process with that commit id but then it doesn't seem to stick. I'm doing something wrong and I can't figure out what.

**Comment by jduranf** [comment]
I'm not a git expert, but maybe you can squash all commits and then sign-off...

File: internal/pkg/db/redis/export.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (andresrinivasan) [reply internal/pkg/db/redis/export.go]: Thanks to @JPWKU for helping me buy a clue. Sign-off completed. Now to separate `.gitignore` and `bin/edgex-docker-launch.sh`

[the remark this answers is not in the document]
Comment (andresrinivasan) [reply internal/pkg/db/redis/export.go]: @jduranf Please see PR #664 and PR #662 for separated `bin/edgex-docker-launch.sh` and `.gitignore`, respectively.

**Comment by tsconn23** [comment]
@andresrinivasan @JPWKU and myself agreed that because of the complications with the commit history of this PR, the UI will be used to merge going forward unless there are changes requested. Yes, this will result in some merge commits.

**Comment by andresrinivasan** [comment]
Superseded by PR #852

**Pull request closed without merging**